File: frontend/src/types/promptVersion.ts
export interface PromptVersion {
  id: number;
  prompt_id: number;
  version: number;

  title: string;
  description: string | null;
  system_prompt: string | null;
  user_prompt: string;

  favorite: boolean;
  collection_id: number | null;

  created_at: string;
}

File: backend/app/schemas/prompt_version.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PromptVersionResponse(BaseModel):
    model_config = ConfigDict(
        from_attributes=True,
    )

    id: int
    prompt_id: int
    version: int

    title: str
    description: str | None
    system_prompt: str | None
    user_prompt: str

    favorite: bool
    collection_id: int | None

    created_at: datetime
